### PR TITLE
Document multi-tenant queries.

### LINKED ISSUE #5993: [Task] Document multi-tenant queries.
Document multi tenant queries for Loki and for GEL, since the way to set it up is slightly different between the two:

- [ ] completed for Loki
- [ ] completed for GEL

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
This change adds documentation for multi-tenant queries. It describes the API usage, limitations and possible filters in queries.

**Which issue(s) this PR fixes**:
Fixes #5993

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [ ] Tests updated
- [ ] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: docs/sources/operations/multi-tenancy.md [hunk 1]
```diff
@@ -17,3 +17,19 @@ To run in multi-tenant mode, Loki should be started with `auth_enabled: true`.
 Loki can be run in "single-tenant" mode where the `X-Scope-OrgID` header is not
 required. In single-tenant mode, the tenant ID defaults to `fake`.
 
+## Multi-tenant Queries
```
Comment (jeschkies) [inline docs/sources/operations/multi-tenancy.md:20]: I wonder whether we should document how to setup a datasource in Grafana for multi-tenant queries. It is not trivial IMHO.

File: docs/sources/operations/multi-tenancy.md [hunk 2]
```diff
@@ -17,3 +17,19 @@ To run in multi-tenant mode, Loki should be started with `auth_enabled: true`.
...
+`multi_tenant_queries_enabled: true` option in the querier. Once enabled multiple
+tenant IDs can be defined in the HTTP header `X-Scope-OrgID` by concatenating them
+with `|`. For instance a query for tenant A and B can set `X-Scope-OrgID: A|B`.
+
+Only query endpoints support multi-tenant calls. Calls to `GET /loki/api/v1/tail`
+and `POST /loki/api/v1/push` will return an HTTP 400 error if more than one tenant
+is defined in the HTTP header.
+
+Instant and range queries support label filtering on the tenant IDs. For example
+`{app="foo", __tenant_id__~="a.+"} | logfmt` will return results for all tenants
```
Comment (xom4ek) [inline docs/sources/operations/multi-tenancy.md:32]:
```suggestion
`{app="foo", __tenant_id__=~"a.+"} | logfmt` will return results for all tenants
```

File: docs/sources/operations/multi-tenancy.md [hunk 3]
```diff
@@ -17,3 +17,19 @@ To run in multi-tenant mode, Loki should be started with `auth_enabled: true`.
 Loki can be run in "single-tenant" mode where the `X-Scope-OrgID` header is not
 required. In single-tenant mode, the tenant ID defaults to `fake`.
 
+## Multi-tenant Queries
+
+If run in multi-tenant mode, queries across different tenants can be enabled via
+`multi_tenant_queries_enabled: true` option in the querier. Once enabled multiple
+tenant IDs can be defined in the HTTP header `X-Scope-OrgID` by concatenating them
+with `|`. For instance a query for tenant A and B can set `X-Scope-OrgID: A|B`.
```
Comment (xom4ek) [inline docs/sources/operations/multi-tenancy.md:25]: That didnt work.
When use `X-Scope-OrgID=onetenant|twotenant` loki response 200 but empty data.

You must use `X-Scope-OrgID` x2 like that:
image

Comment (jeschkies) [reply docs/sources/operations/multi-tenancy.md:25]: Inserting. The piping worked for me. Let me investigate. Thanks for trying it out.

Comment (jeschkies) [reply docs/sources/operations/multi-tenancy.md:25]: Ok, I've tried it with the Docker compose setup in `tools/dev/loki-boltdb-storage-s3`. How did you try it out?

File: docs/sources/operations/multi-tenancy.md [hunk 4]
```diff
@@ -17,3 +17,19 @@ To run in multi-tenant mode, Loki should be started with `auth_enabled: true`.
...
+
+Only query endpoints support multi-tenant calls. Calls to `GET /loki/api/v1/tail`
+and `POST /loki/api/v1/push` will return an HTTP 400 error if more than one tenant
+is defined in the HTTP header.
+
+Instant and range queries support label filtering on the tenant IDs. For example
+`{app="foo", __tenant_id__=~"a.+"} | logfmt` will return results for all tenants
+whose ID stat with `a`. Tenant ID filter in stages is not supported `{app="foo"} | __tenant_id__="1" | logfmt` will not work.
```
Comment (JordanRushing) [inline docs/sources/operations/multi-tenancy.md:33]:
```suggestion
whose ID stat with `a`. Tenant ID filtering in stages is not supported; `{app="foo"} | __tenant_id__="1" | logfmt` will not work.
```

**Review by JordanRushing — APPROVED** [review]

File: docs/sources/operations/multi-tenancy.md (hunk 3 above)
Comment (xom4ek) [reply docs/sources/operations/multi-tenancy.md:25]:
```yaml
analytics:
  reporting_enabled: false
auth_enabled: true
common:
  path_prefix: /var/loki
  replication_factor: 1
  storage:
    s3:
      access_key_id: access
      bucketnames: bucketname
      endpoint: endpoint
      http_config:
        idle_conn_timeout: 90s
        insecure_skip_verify: false
        response_header_timeout: 0s
      region: region
      secret_access_key: secret
compactor:
  shared_store: aws
  working_directory: /var/loki/compactor
frontend:
  log_queries_longer_than: 20s
memberlist:
  join_members:
  - loki-memberlist
querier:
  multi_tenant_queries_enabled: true
schema_config:
  configs:
  - from: "2021-09-07"
    index:
      period: 24h
      prefix: loki_index_
    object_store: aws
    schema: v11
    store: boltdb-shipper
storage_config:
  aws:
    access_key_id: access
    bucketnames: bucketname
    endpoint: endpoint
    http_config:
      idle_conn_timeout: 90s
      insecure_skip_verify: false
      response_header_timeout: 0s
    region: region
    secret_access_key: secret
  boltdb_shipper:
    active_index_directory: /var/loki/index
    cache_location: /var/loki/index_cache
    shared_store: s3
tracing:
  enabled: true
```

So i found that when use simple-scale helm chart.


image

image

If you want i cant provide logs from grafana for answers loki.

Comment (xom4ek) [reply docs/sources/operations/multi-tenancy.md:25]: Seems like on my instance multi-tenant queries didnt work.
I need send __tenant_id__ from log collector? Or loki create that index selfs?

Comment (jeschkies) [reply docs/sources/operations/multi-tenancy.md:25]: What are the screenshots showing? What are tags here? `X-Scope-OrgID='default|system'
image` should be correct on the datasource. However, that also means Promtail is sending logs either as `default` or `system` tenant.

Comment (xom4ek) [reply docs/sources/operations/multi-tenancy.md:25]: Let me explain.


When i use `Loki` datasource and set `X-Scope-OrgID=default` or `X-Scope-OrgID=system` - we get that graph:
image


But when i set `X-Scope-OrgID=default|system` i get "no data".

So my first screenshots just trace from loki where read service get 0 Series when `tenants_ids=[default|system]` and 3k when `tenants_ids=[default]` 



About promtail - already have tenants in loki and `X-Scope-OrgID` without `multitenant_query` work fine - just in code and your example i see `__tenant_id` label and didnt see that in my loki installation

Comment (jeschkies) [reply docs/sources/operations/multi-tenancy.md:25]: Thanks for your patience. I'm not really sure where you see `tenants_ids=[default|system]`.

Anyhow, which version of Loki are you using? It must be a recent main build after at least commit `[c7cbeab](https://github.com/grafana/loki/commit/c7cbeab8e2338b1dc39a3ff3bcde29af58317823)`.

File: docs/sources/operations/multi-tenancy.md [hunk 5]
```diff
@@ -17,3 +17,19 @@ To run in multi-tenant mode, Loki should be started with `auth_enabled: true`.
...
+tenant IDs can be defined in the HTTP header `X-Scope-OrgID` by concatenating them
+with `|`. For instance a query for tenant A and B can set `X-Scope-OrgID: A|B`.
+
+Only query endpoints support multi-tenant calls. Calls to `GET /loki/api/v1/tail`
+and `POST /loki/api/v1/push` will return an HTTP 400 error if more than one tenant
+is defined in the HTTP header.
+
+Instant and range queries support label filtering on the tenant IDs. For example
+`{app="foo", __tenant_id__=~"a.+"} | logfmt` will return results for all tenants
+whose ID stat with `a`. Tenant ID filtering in stages is not supported; `{app="foo"} | __tenant_id__="1" | logfmt` will not work.
```
Comment (KMiller-Grafana) [inline docs/sources/operations/multi-tenancy.md:33]:
```suggestion
whose ID start with `a`. Tenant ID filtering in stages is not supported; `{app="foo"} | __tenant_id__="1" | logfmt` will not work.
```

File: docs/sources/operations/multi-tenancy.md [hunk 6]
```diff
@@ -17,3 +17,19 @@ To run in multi-tenant mode, Loki should be started with `auth_enabled: true`.
 Loki can be run in "single-tenant" mode where the `X-Scope-OrgID` header is not
 required. In single-tenant mode, the tenant ID defaults to `fake`.
 
+## Multi-tenant Queries
+
+If run in multi-tenant mode, queries across different tenants can be enabled via
+`multi_tenant_queries_enabled: true` option in the querier. Once enabled multiple
```
Comment (KMiller-Grafana) [inline docs/sources/operations/multi-tenancy.md:23]: I'm trying for better wording than "queries across different tenants." What do you think?

```suggestion
If run in multi-tenant mode, queries that gather results from multiple tenants can be enabled with
the `multi_tenant_queries_enabled: true` configuration option in the querier. Once enabled, multiple
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]

File: docs/sources/operations/multi-tenancy.md (hunk 6 above)
Comment (jeschkies) [reply docs/sources/operations/multi-tenancy.md]: That is much cleaner. Thanks. I'm wondering though if the results a from or for multiple tenants but that would be nit picking :slightly_smiling_face:

File: docs/sources/operations/multi-tenancy.md (hunk 3 above)
Comment (xom4ek) [reply docs/sources/operations/multi-tenancy.md:25]: Yes - i found problem.

We didnt update master branch after release.
So right now everything fine.
Thanks for help!

**Review by KMiller-Grafana — APPROVED** [review]
Docs LGTM.

**Pull request merged**